lightning/src/ln/peer_handler.rs

```diff
@@ -33,22 +33,131 @@ use routing::network_graph::NetGraphMsgHandler;
+		DummyChannelHandler::push_error(self, their_node_id, msg.channel_id);
+	}
+	fn handle_funding_locked(&self, their_node_id: &PublicKey, msg: &msgs::FundingLocked) {
+		DummyChannelHandler::push_error(self, their_node_id, msg.channel_id);
+	}
+	fn handle_shutdown(&self, their_node_id: &PublicKey, _their_features: &InitFeatures, msg: &msgs::Shutdown) {
+		DummyChannelHandler::push_error(self, their_node_id, msg.channel_id);
+	}
+	fn handle_closing_signed(&self, their_node_id: &PublicKey, msg: &msgs::ClosingSigned) {
+		DummyChannelHandler::push_error(self, their_node_id, msg.channel_id);
+	}
+	fn handle_update_add_htlc(&self, their_node_id: &PublicKey, msg: &msgs::UpdateAddHTLC) {
+		DummyChannelHandler::push_error(self, their_node_id, msg.channel_id);
+	}
+	fn handle_update_fulfill_htlc(&self, their_node_id: &PublicKey, msg: &msgs::UpdateFulfillHTLC) {
+		DummyChannelHandler::push_error(self, their_node_id, msg.channel_id);
+	}
+	fn handle_update_fail_htlc(&self, their_node_id: &PublicKey, msg: &msgs::UpdateFailHTLC) {
+		DummyChannelHandler::push_error(self, their_node_id, msg.channel_id);
+	}
+	fn handle_update_fail_malformed_htlc(&self, their_node_id: &PublicKey, msg: &msgs::UpdateFailMalformedHTLC) {
+		DummyChannelHandler::push_error(self, their_node_id, msg.channel_id);
+	}
+	fn handle_commitment_signed(&self, their_node_id: &PublicKey, msg: &msgs::CommitmentSigned) {
+		DummyChannelHandler::push_error(self, their_node_id, msg.channel_id);
+	}
+	fn handle_revoke_and_ack(&self, their_node_id: &PublicKey, msg: &msgs::RevokeAndACK) {
+		DummyChannelHandler::push_error(self, their_node_id, msg.channel_id);
+	}
+	fn handle_update_fee(&self, their_node_id: &PublicKey, msg: &msgs::UpdateFee) {
+		DummyChannelHandler::push_error(self, their_node_id, msg.channel_id);
+	}
+	fn handle_announcement_signatures(&self, their_node_id: &PublicKey, msg: &msgs::AnnouncementSignatures) {
+		DummyChannelHandler::push_error(self, their_node_id, msg.channel_id);
+	}
+	fn peer_disconnected(&self, _their_node_id: &PublicKey, _no_connection_possible: bool) {}
+	fn peer_connected(&self, _their_node_id: &PublicKey, _msg: &msgs::Init) {}
+	fn handle_channel_reestablish(&self, their_node_id: &PublicKey, msg: &msgs::ChannelReestablish) {
+		DummyChannelHandler::push_error(self, their_node_id, msg.channel_id);
+	}
+	fn handle_error(&self, _their_node_id: &PublicKey, _msg: &msgs::ErrorMessage) {}
+}
+impl Deref for DummyChannelHandler {
+	type Target = DummyChannelHandler;
+	fn deref(&self) -> &Self { self }
+}
+
 /// Provides references to trait impls which handle different types of messages.
 pub struct MessageHandler<CM: Deref, RM: Deref> where
 		CM::Target: ChannelMessageHandler,
 		RM::Target: RoutingMessageHandler {
 	/// A message handler which handles messages specific to channels. Usually this is just a
-	/// ChannelManager object.
+	/// ChannelManager object or a DummyChannelHandler.
```

Comment:
This comment applies to `PeerManager::new(..)` function:
could we make `message_handler` an optional parameter to that function, and insert these two dummies if it's `None`?

Reply:
You probably never actually want to create a MessageHandler with both set to dummy values...you won't ever *do* anything. Maybe I misunderstood your comment?

Reply:
Oh oops true. Revised suggestion: could we make `PeerManager::new(..)` be parameterized by a `CM` and an `RM` separately, and have the `RM` be optional?

Reply:
I actually think both are likely, its totally reasonable to run a thing to fetch the routing graph.